R/req-url.R

```diff
@@ -67,14 +64,20 @@ req_url_path <- function(req, ...) {
 req_url_path_append <- function(req, ...) {
   check_request(req)
   path <- paste(..., sep = "/")
+  path <- path_add_slash(path)
 
   url <- url_parse(req$url)
 
+  url$path <- paste0(sub("/$", "", url$path), path)
+
+  req_url(req, url_build(url))
+}
+
+path_add_slash <- function(path) {
   # Ensure we don't add duplicate /s
-  if (!grepl("^/", path)) {
+  if (!is_empty(path) && !grepl("^/", path)) {
```

Comment:
This adds special handling for the length-0 case, which makes me think we should also consider the length >1 case, which suggests that maybe instead of 

```R
  path <- paste(..., sep = "/")
```

we need

```R
  path <- paste(..., sep = "/", collapse = "/")
```

which will ensure that `path` is always a length-1 character vector and simplify this code path.

Reply:
Hmm, then this would work

``` r
req <- request("http://example.com/x")
req_url_path_append(req, c("a", "b"), "c")
#> <httr2_request>
#> GET http://example.com/x/a/c/b/c
#> Body: empty
```

<sup>Created on 2022-10-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

which feels a bit weird to me. The alternative would be to check the size of path (empty or length 1). What do you prefer?

Reply:
It's nice to allow this though:

```R
pieces <- c("a", "b", "c")
req <- request("http://example.com/x")
req_url_path_append(req, pieces)
```

Reply:
Or support splicing?

```r
pieces <- c("a", "b", "c")
req <- request("http://example.com/x")
req_url_path_append(req, !!!pieces)
```

Reply:
IMO splicing seems overkill when the input is an atomic vector because `c(c(x), c(y))` is the same as `c(x, y)`